docker/scripts/cyber_start.sh

```diff
@@ -15,265 +15,509 @@
-    cp $AGREEMENT_FILE $agreement_record
-    echo "$tip" >> $agreement_record
-    echo "$user_agreed" >> $agreement_record
-  else
-    exit 1
-  fi
+    local agreement_record="${HOME}/.apollo_agreement.txt"
+    if [[ -e "${agreement_record}" ]]; then
+        return 0
+    fi
+    local agreement_file
+    agreement_file="$APOLLO_ROOT_DIR/scripts/AGREEMENT.txt"
+    if [[ ! -f "${agreement_file}" ]]; then
+        error "AGREEMENT ${agreement_file} does not exist."
+        exit 1
+    fi
+
+    cat "${agreement_file}"
+    local tip="Type 'y' or 'Y' to agree to the license agreement above, \
+or type any other key to exit:"
+    echo -n "${tip}"
+    read -r -n 1 user_agreed
+    echo
+    if [[ "${user_agreed}" == "y" || "${user_agreed}" == "Y" ]]; then
+        cp -f "${agreement_file}" "${agreement_record}"
+        echo
+        echo "${tip}" >> "${agreement_record}"
+        echo "${user_agreed}" >> "${agreement_record}"
+    else
+        exit 1
+    fi
 }
 
 function check_host_environment() {
-  echo 'Host environment checking done.'
+    echo "Done checking host environment."
+}
+
+function _optarg_check_for_opt() {
+    local opt="$1"
+    local optarg="$2"
+
+    if [[ -z "${optarg}" || "${optarg}" =~ ^-.* ]]; then
+        error "Missing parameter for ${opt}. Exiting..."
+        exit 2
+    fi
+}
+
+function _target_arch_check() {
+    local arch="$1"
+    if [[ "${SUPPORTED_ARCHS}" != *" ${arch} "* ]]; then
+        error "Unsupported target architecture: ${arch}. Allowed values:${SUPPORTED_ARCHS}"
+        exit 1
+    fi
 }
 
-function show_usage()
-{
+function show_usage() {
 cat <<EOF
-Usage: $(basename $0) [options] ...
+Usage: $0 [options] ...
 OPTIONS:
-    -C                     Pull docker image from China mirror.
+    -g <us|cn>             Pull docker image from mirror registry based on geolocation.
     -h, --help             Display this help and exit.
     -t, --tag <version>    Specify which version of a docker image to pull.
     -l, --local            Use local docker image.
-    stop                   Stop all running Apollo containers.
+    -m <arch>              Specify docker image for a different CPU arch.
+    stop [-f|--force]      Stop all running Apollo containers. Use "-f" to force removal.
 EOF
-exit 0
 }
 
-function stop_containers()
-{
-running_containers=$(docker ps --format "{{.Names}}")
-
-for i in ${running_containers[*]}
-do
-  if [[ "$i" =~ apollo_* ]];then
-    printf %-*s 70 "stopping container: $i ..."
-    docker stop $i > /dev/null
-    if [ $? -eq 0 ];then
-      printf "\033[32m[DONE]\033[0m\n"
+function stop_all_apollo_containers_for_user() {
+    local force="$1"
+    local running_containers
+    running_containers="$(docker ps -a --format '{{.Names}}')"
+    for container in ${running_containers[*]} ; do
+        if [[ "${container}" =~ apollo_.*_${USER} ]] ; then
+            #printf %-*s 70 "Now stop container: ${container} ..."
+            #printf "\033[32m[DONE]\033[0m\n"
```

Comment:
Remove only apollo containers created by $USER himself. Useful in a workstation situation.